### ISSUE TITLE
Questionable physics with destroyed submarine walls (Or just entering submarine in general)

### ISSUE THREAD

**Issue by NilanthAnimosus** [Regalis11/Barotrauma#9564]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

I destroyed curved/diagonal wall pieces of a submarine, and attempted to swim through from the top and was blocked.

However I can swim through just fine from the side where then it shows it doesn't seem to be a solid wall blocking me, as shown in this short mp4:
https://i.gyazo.com/91076b235a23c3b5c67fa3e7246b84b9.mp4

Additionally, swimming in from the side and down does let me teleport through the wall essentially and get stuck within the submarines walls (when behind the wall im using the mouse to show where im trying to swim):

https://i.gyazo.com/ae89eb09736c2503faa84867c67ca702.mp4

Swimming in diagonally when it is completely destroyed is fine, swimming straight down is a wall. and im quite certain its fully damaged while not letting me pass:

![image](https://user-images.githubusercontent.com/29177976/178844514-172c474d-ab66-48c0-9109-491cae26035c.png)

I feel submarine wall collision issues (and crossing from the ocean to the submarine world position) used to be an old issue but I couldn't find an issue that seemed like this.

### Reproduction steps

I got this while checking a submarine in the submarine editor - but I feel it will occur anywhere.

As an additional sidenote, the submarine in question was the azimuth

I was originally using melee weapons, but noticing this began to use explosion commands to fully destroy the walls to ensure the wall was fully damaged.

### Bug prevalence

Happens every time I play

### Version

Other

### -

0.18.15.0 unstable

### Which operating system did you encounter this bug on?

Windows

**Comment by NilanthAnimosus** [comment]
Tested against outside-collision-fix commit https://github.com/Regalis11/Barotrauma-development/commit/6c0d48c0eee20a6a89bdd08b02071654433044e6

works for going vertically or horizontally into a broken piece, but I noticed something strange best shown on video:

https://user-images.githubusercontent.com/29177976/186886287-a5eabaae-fb5b-4298-b14c-22e0ed4e7bfa.mp4

essentially if you begin by going down/sideways into a none-broken diagonal piece of a wall, you will glide along it and not pass through even if your tapping the key along the way over holding. you must swim away then swim back to pass through. I find this pretty unusual for behavior as I am basically following a hole like its a physical wall unable to pass through until I swim away.
